### ISSUE TITLE
[Bug] Halting Error in DevNet on Mining

### ISSUE THREAD

**Issue by lpgeiger** [OLSF/libra#440]
TLDR; this may be specific to devnet configs. However it is a networking error related to onchain reconfigs, which happens after epoch reconfigurations.

### Scenario
Alice Bob and carol, are validating and mining normally (CASE 1)

Eve joins the network, after Alice submits a create-validator transaction.

Divergence from mainnet:  in devnet all elegible validators go into set (if there are 3 or less CASE 1 validators). So, Eve goes directly into the validator-set  without ever having voted.

## Problem

If Eve does not vote (e.g. is a full node).

AND

If Eve submits mining transactions.

... The next epoch will halt on round 1 (though the block prologue runs normally).

The errors on halting are Network connection errors.

## Notes

In swarm when tested with 10 nodes, it took a few blocks before such halting happened.

If Eve votes before mining, then there is no issue, network goes on normally. The issue only happens if Eve mines after voting.

Functional test cases simulating the scenario don't produce errors.  Using branch `lg-debug` these are called `debug_onboarding_case_3.move` and `debug_case_3.move`

## Reproduction

Using branch `lg-debug`

start a swarm locally: 
```
cd ~/libra/ 
cargo build -p libra-node -p cli && cargo run -p libra-swarm -- --libra-node target/debug/libra-node -c ~/libra/swarm_temp -n 5 -s --cli-path target/debug/cli
```

onboard `eve`
```
cd ~/libra/
cargo r -p txs -- --swarm-path ./swarm_temp/ create-validator -f ./fixtures/onboarding/eve_init_test.json 
```

mine against swarm with the `eve` persona
```
cd ~/libra/
cargo r -p miner -- swarm --persona eve

```

Errors:

Lots of errors like this:

>```[consensus] WARN consensus/src/epoch_manager.rs:579 {"committed_round":0,"error":"Round 1 timeout, broadcast to all peers","kind":"InternalError","pending_votes":"PendingVotes: [LI 29f51f58 has 4 votes [284C635FB67AF3D5CFDBAE1EE8930009, 4C613C2F4B1E67CA8D98A542EE3F59F5, 5DFA1B987A420C0FE23A6F3A8F27B7CA, 975B47350701D8C5563BF5946F7BB986] LI 59e7b59c has 1 votes [5709AFA1D0FBB40B171AA03D457DB72A] 5 timeout [284C635FB67AF3D5CFDBAE1EE8930009, 4C613C2F4B1E67CA8D98A542EE3F59F5, 5709AFA1D0FBB40B171AA03D457DB72A, 5DFA1B987A420C0FE23A6F3A8F27B7CA, 975B47350701D8C5563BF5946F7BB986]]","round":1,"self_vote":"Vote: [vote data: VoteData: [block id: f081fe15, epoch: 16, round: 01, parent_block_id: 939d5c69, parent_block_round: 00], author: 4c613c2f, is_timeout: true, LedgerInfo: [commit_info: BlockInfo: [epoch: 0, round: 0, id: 00000000, executed_state_id: 00000000, version: 0, timestamp (us): 0, next_epoch_state: None]]]"}```

>```2021-04-10T00:12:57.132437Z [consensus] DEBUG consensus/src/round_manager.rs:624 Vote: [vote data: VoteData: [block id: f081fe15, epoch: 16, round: 01, parent_block_id: 939d5c69, parent_block_round: 00], author: 4c613c2f, is_timeout: true, LedgerInfo: [commit_info: BlockInfo: [epoch: 0, round: 0, id: 00000000, executed_state_id: 00000000, version: 0, timestamp (us): 0, next_epoch_state: None]]] {"epoch":16,"event":"ReceiveVote","remote_peer":"4C613C2F4B1E67CA8D98A542EE3F59F5","round":1}```

**Comment by lpgeiger** [comment]
@sm86 Is it possible that the weight of consensus gets skewed in the devnet, and the only validator mining has 90% + of the weight, and no one else gets to propose?

**Comment by sm86** [comment]
No, the proposals are not dependent on mining. Need to confirm whether it is random or depends on node weight, either way it is not updated in real time

**Comment by sm86** [comment]
We've had situations when network ran with 10 nodes and one of them is still a full node and mines, it never halted. 

So, I'd assume it might have something to do with a new node possibly

**Comment by lpgeiger** [comment]
Just finished some tests.  It looks like if in a validator set of 10, only one validator is mining, then it will monopolize the proposals. My hypothesis is that if the only node mining (and monopolizing proposals in a new epoch) is offline it will lead to a network halt. If this is true it's an extreme edge case.

<img width="533" alt="Screen Shot 2021-04-09 at 10 39 48 PM" src="https://user-images.githubusercontent.com/1364012/114255913-7f92ec00-9984-11eb-83b1-d60ea89eedcb.png">

**Comment by Albe540** [comment]
I was trying to mint NFT on moonstream and I received this message:
"Sign Transaction
www.moonstream.com is requesting to sign the following transaction

Transaction simulation:

Failed to simulate transaction on Devnet"

Why?????????
